autoload/deoplete/handlers.vim

```diff
@@ -51,6 +51,8 @@ function! s:completion_begin(event) abort "{{{
   " Save the previous position
   let g:deoplete#_context.position = context.position
 
+  call deoplete#mappings#_set_completeopt()
```

Comment:
I needed this change, because otherwise it would select and insert the first entry automatically.

I am using `completeopt=longest,menuone` by default.

I have `context.filetypes = ['python', 'python']` here, and the `feedkeys("\<C-x>\<C-o>", 'n')` does not get triggered.  And "python" is not in `g:deoplete#_omni_patterns`, which is `{'mkd': ['<[^>]*'], 'xhtml': ['<[^>]*'], 'go': ['[^. \t0-9]\.\w*'], 'markdown': ['<[^>]*'], '
xml': ['<[^>]*'], 'c': ['[^. \t0-9]\.\w*', '[^. \t0-9]->\w*'], 'html': ['<[^>]*'], 'java': ['
[^. \t0-9]\.\w*'], 'cpp': ['[^. \t0-9]\.\w*', '[^. \t0-9]->\w*', '[a-zA-Z_]\w*::\w*']}`.

So then `call rpcnotify(g:deoplete#_channel_id, 'completion_begin', context)` gets triggered, and s:completion_begin will call `let context = deoplete#init#_context(a:event, [])` - overwriting the saved completeopt.

I think the problem is that completeopt cannot be set in an `<c-r>=`-mapping just before calling `complete()`?! https://github.com/blueyed/deoplete.nvim/blob/6e8a4c8fa33f73a8bdbcd5e93453ea5809cb39af/autoload/deoplete/mappings.vim#L26-L34